core/src/main/java/feign/Contract.java

```diff
@@ -211,99 +219,237 @@ protected void nameParam(MethodMetadata data, String name, int i) {
     }
   }
 
-  class Default extends BaseContract {
+  /**
+   * {@link Contract} base implementation that works by declaring witch annotations should be
+   * processed and how each annotation modifies {@link MethodMetadata}
+   */
+  public abstract class DeclarativeContract extends BaseContract {
```

Comment:
note to self:
Extract this class to a new source file before merging.  Enough is enough =)